### DISCUSSION TITLE
g++ not available (Windows 10)

### DISCUSSION THREAD

**Discussion by JeanLuc001** [pymc-devs/pymc3#6763] [Q&A]
Hi,

I keep getting the following warnings when importing PyMC in a Jupyter notebook under VSCode:

> WARNING (pytensor.configdefaults): g++ not available, if using conda: `conda install m2w64-toolchain`
> WARNING (pytensor.configdefaults): g++ not detected!  PyTensor will be unable to compile C-implementations and will default to Python. Performance may be severely degraded. To remove this warning, set PyTensor flags cxx to an empty string.
> WARNING (pytensor.tensor.blas): Using NumPy C-API based implementation for BLAS functions.
> Running on PyMC v5.4.1

I followed the PyMC installation instructions at

https://www.pymc.io/projects/docs/en/stable/installation.html

and even tried to install g++ (v13.1.0) via msys64 on my own after finding the following discussion:

https://discourse.pymc.io/t/getting-a-working-pymc3-windows-10-installation-using-anaconda-an-installation-guide/8641

Anaconda version: 2.3.2
conda version: 23.5.0

**Comment by shixinghari** [comment]
I have the same problem when I (import pymc) in pycharm and jupyter,
The error:
WARNING (pytensor.configdefaults): g++ not available, if using conda: `conda install m2w64-toolchain`
WARNING (pytensor.configdefaults): g++ not detected!  PyTensor will be unable to compile C-implementations and will default to Python. Performance may be severely degraded. To remove this warning, set PyTensor flags cxx to an empty string.
WARNING (pytensor.tensor.blas): Using NumPy C-API based implementation for BLAS functions.
UnicodeDecodeError: 'gbk' codec can't decode byte 0xba in position 2367: illegal multibyte sequence

And I have solved it! Here is my method:
create a new evironment in conda, choose python 3.6, then install pymc, everything goes totally different than python3.10.
seems the version of python is the key.

**Reply by ricardoV94** [reply]
Newer versions of PyMC aren't compatible with Python 3.6 (nor is much of the scientific ecosystem), so you'll be missing on new features and bugfixes.

**Reply by shixinghari** [reply]
Thanks to your advice!

I run (conda create -c conda-forge -n pymc_env "pymc>=5"
conda activate pymc_env) in Anaconda prompt, instead of the GUI

But new evironment 'pymc_env' with python3.11 and pymc 5.6.1 and arviz 0.16.1 can't (import pymc), the same error. 

I think the problem is because (arviz), I get the similiar error when I (import arviz)
error:
Traceback (most recent call last):
  File "C:\Users\toki\Desktop\Py\格式化输入方法.py", line 3, in <module>
    import arviz
  File "E:\MEDownload\anacondada\envs\pymc_env\Lib\site-packages\arviz\__init__.py", line 33, in <module>
    from .data import *
(balabala)
UnicodeDecodeError: 'gbk' codec can't decode byte 0xba in position 2367: illegal multibyte sequence

my system is Window10
In python 3.6, I can only use the version 2.3.8, but seem enough for me to learn Bayesian Method For Hackers.

**Comment by shixinghari** [comment]
I don't know why, but just use (conda create -c conda-forge -n pymc_env "pymc=5.7.2") in conda prompt will make this problem disappear (now the newest version is 5.7.2). 
And everything is fine except my jupyter notebook, it get error(internal serve 500). Following the advice, I update the (nbconvert), then jupyter is OK, but the same error occur when I (import pymc).
Definetely, I'm idiot(´-ι_-｀).
